File: Haskell/grade_calc.hs
import Data.List (intercalate)

-- Calculate the letter grade based on average score
calculateGrade :: Double -> Char
calculateGrade avg
    | avg >= 90 = 'A'
    | avg >= 80 = 'B'
    | avg >= 70 = 'C'
    | avg >= 60 = 'D'
    | otherwise = 'F'

-- Calculate average of three scores
calculateAverage :: Double -> Double -> Double -> Double
calculateAverage math science english = (math + science + english) / 3

-- Process a single student's grades
processStudent :: String -> Double -> Double -> Double -> String
processStudent name math science english =
    let avg = calculateAverage math science english
        grade = calculateGrade avg
    in name ++ ": Average = " ++ show avg ++ ", Grade = " ++ [grade]

-- Main function with multiple students
main :: IO ()
main = do
    putStrLn "=== Student Grade Calculator ==="
    putStrLn ""
    
    -- Example: Process multiple students
    let students = 
            [ ("Alice", 85, 90, 88)
            , ("Bob", 92, 95, 89)
            , ("Charlie", 78, 75, 82)
            , ("Diana", 65, 70, 68)
            ]
    
    -- Map the processStudent function over all students
    let results = map (\(name, math, science, english) -> 
                        processStudent name math science english) 
                      students
    
    -- Print all results
    mapM_ putStrLn results
    
    putStrLn ""
    putStrLn "Thank you for using the calculator!"

File: Lua/grade_calc.lua
function calculateGrade(average)
    if average >= 90 then
        return "A"
    elseif average >= 80 then
        return "B"
    elseif average >= 70 then
        return "C"
    elseif average >= 60 then
        return "D"
    else
        return "F"
    end
end

function main()
    while true do
        print("=== Student Grade Calculator ===")
        print("")
        
        io.write("Enter student name: ")
        local name = io.read()
        
        io.write("Enter math score (0-100): ")
        local math = tonumber(io.read())
        
        io.write("Enter science score (0-100): ")
        local science = tonumber(io.read())
        
        io.write("Enter english score (0-100): ")
        local english = tonumber(io.read())
        
        print("")
        
        local average = (math + science + english) / 3
        local grade = calculateGrade(average)
        
        print("Name: " .. name)
        print(string.format("Average: %.2f", average))
        print("Grade: " .. grade)
        print("")
        
        io.write("Calculate another student? (Y/N): ")
        local response = io.read()
        
        if response ~= "Y" and response ~= "y" then
            break
        end
    end
    
    print("Thank you for using the calculator!")
end

main()
